docs/blog/2020/CPP-guide.md

```diff
@@ -0,0 +1,279 @@
+- `xref-find-references`(`s-l g r`) - find references to the symbol at point
+![References](images/images/references-helm.png "References helm")
+- `helm-imenu` - browse the symbols in current document
+- `helm-lsp-workspace-symbol` - find symbol in current project
+![workspace symbols](images/workspace-symbols.png "workspace symbols")
+- `helm-lsp-global-workspaces-symbol` - find symbol in all projects
+- `lsp-treemacs-type-hierarchy` - show type hierarchy
+
+_Note:_ if you prefer [ivy](https://github.com/abo-abo/swiper) over `helm` you may check out [lsp-ivy](https://github.com/emacs-lsp/lsp-ivy).
+
+### [lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs)
+`lsp-treemacs` is a package providing integration with
+[treemacs](https://github.com/Alexander-Miller/treemacs) and an alternative
+tree-view visualization. Refer to the project's readme for further information.
+
+![tree references](images/treemacs-references.png "Treemacs references")
+
+### Help/Documentation
+`lsp-mode` automatically enables `eldoc-mode` which will show hover information
+in the minibuffer. `lsp-mode` is showing only the one line signature info so if
+you want to see the full documentation at point you can use `lsp-describe-thing-at-point`.
+
+![hover](images/hover-info.png)
+
+In addition to that you can also see signature help after pressing `C-M-SPC` or
+after pressing trigger char like `(`. If there is more than one applicable signature due to function overloading, you may browse between the available candidates using `M-n/M-p`.
+may browse them via `M-n/M-p`
+
+![signature info](images/signature-help.png)
+
+### Diagnostics
+For on-the-fly errors `lsp-mode` is using `flycheck` (`flymake` is also
+supported). It is configured automatically. In addition to standard `flycheck`
+features `lsp-mode` provides a project-wide error list via
+`lsp-treemacs-errors-list`.
+
+![errors list](images/errors-list.png)
+
+## Debugging
+
+Just like `lsp-mode` provides editing features through the editor-independent [Language Server Protocol](https://microsoft.github.io/language-server-protocol/), its sister package `dap-mode` provides debugging features through the editor-independent [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/) and language-specific debugging servers called _debug adapters_.
+front-end of the sister [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/). It is again client server
+like `Language Sever Protocol`. `dap-mode` provides all of the traditional
+debugger features - breakpoints(conditions, hit count, etc), threads, locals,
+
+### Installation
+
+`dap-mode` provides installation commands for most of the debug adapters. In this
+tutorial we will use the [vscode-cpptools](https://github.com/microsoft/vscode-cpptools) debug adapter. To install the adapter
+do `M-x dap-cpptools-setup`.
+
+_Note:_ you should have `gdb` on path.
```

Comment:
For macOS users, `lldb` is more popular than `gdb` (which is not available in the base system). As the debug adapter supports both I can expand the guide later to show how it works with `lldb`.

Reply:
Yes. I was about to suggest someone using macOS(or other popular OS-es) to provide the macos specific steps.